packages/chrono-mongo-datastore/src/chrono-mongo-datastore.ts

```diff
@@ -48,27 +48,51 @@ export class ChronoMongoDatastore<TaskMapping extends TaskMappingBase>
   implements Datastore<TaskMapping, MongoDatastoreOptions>
 {
   private config: ChronoMongoDatastoreConfig;
-  private database: Db;
+  private database: Db | undefined;
+  private databaseResolvers: Array<(database: Db) => void> = [];
 
-  private constructor(database: Db, config?: Partial<ChronoMongoDatastoreConfig>) {
-    this.database = database;
+  constructor(config?: Partial<ChronoMongoDatastoreConfig>) {
     this.config = {
       completedDocumentTTLSeconds: config?.completedDocumentTTLSeconds,
       collectionName: config?.collectionName || DEFAULT_COLLECTION_NAME,
     };
   }
 
-  static async create<TaskMapping extends TaskMappingBase>(
-    database: Db,
-    config?: Partial<ChronoMongoDatastoreConfig>,
-  ): Promise<ChronoMongoDatastore<TaskMapping>> {
-    const datastore = new ChronoMongoDatastore<TaskMapping>(database, config);
+  /**
+   * Sets the database connection for the datastore. Ensures that the indexes are created and resolves any pending promises waiting for the database.
+   *
+   * @param database - The database to set.
+   */
+  async setDatabase(database: Db) {
+    if (this.database) {
+      throw new Error('Database connection already set');
+    }
 
-    await ensureIndexes(database.collection(datastore.config.collectionName), {
-      expireAfterSeconds: datastore.config.completedDocumentTTLSeconds,
+    this.database = database;
+
+    await ensureIndexes(database.collection(this.config.collectionName), {
```

Comment:
this is going to sound stupid but can i get you to ensureIndexes before setting the DB instance class

Reply:
will just ensure indexes are set before any tasks can be scheduled